python/cudf/cudf/tests/test_avro_reader_fastavro_integration.py

```diff
@@ -244,3 +245,115 @@ def test_avro_compression(rows, codec):
     got_df = cudf.read_avro(buffer)
 
     assert_eq(expected_df, got_df)
+
+
+avro_logical_type_params = [
+    # (avro logical type, avro primitive type, cudf expected dtype)
+    ("date", "int", "datetime64[s]"),
+]
+
+
+@pytest.mark.parametrize(
+    "logical_type, primitive_type, expected_dtype", avro_logical_type_params
+)
+@pytest.mark.parametrize("namespace", [None, "root_ns"])
+@pytest.mark.parametrize("nullable", [True, False])
+@pytest.mark.parametrize("prepend_null", [True, False])
+def test_can_detect_dtypes_from_avro_logical_type(
+    logical_type,
+    primitive_type,
+    expected_dtype,
+    namespace,
+    nullable,
+    prepend_null,
+):
+    avro_type = [{"logicalType": logical_type, "type": primitive_type}]
+    if nullable:
+        if prepend_null:
+            avro_type.insert(0, "null")
+        else:
+            avro_type.append("null")
+
+    schema = fastavro.parse_schema(
+        {
+            "type": "record",
+            "name": "test",
+            "namespace": namespace,
+            "fields": [{"name": "prop", "type": avro_type}],
+        }
+    )
+
+    actual = cudf_from_avro_util(schema, [])
+
+    expected = cudf.DataFrame(
+        {"prop": cudf.Series(None, None, expected_dtype)}
+    )
+
+    assert_eq(expected, actual)
+
+
+def get_days_from_epoch(date: Optional[datetime.date]) -> Union[int, None]:
+    if date is None:
+        return None
+    return (date - datetime.date(1970, 1, 1)).days
+
+
+@pytest.mark.parametrize("namespace", [None, "root_ns"])
+@pytest.mark.parametrize("nullable", [True, False])
+@pytest.mark.parametrize("prepend_null", [True, False])
+def test_can_parse_avro_date_logical_type2(namespace, nullable, prepend_null):
+
+    avro_type = {"logicalType": "date", "type": "int"}
+    if nullable:
+        avro_type = [avro_type]
+        if prepend_null:
+            avro_type.insert(0, "null")
+        else:
+            avro_type.append("null")
+
+    schema_dict = {
+        "type": "record",
+        "name": "test",
+        "fields": [
+            {"name": "o_date", "type": avro_type},
+        ],
+    }
+
+    if namespace:
+        schema_dict["namespace"] = namespace
+
+    schema = fastavro.parse_schema(schema_dict)
+
+    if nullable:
+        dates = [
+            datetime.date(1970, 1, 1),
+            None,
+            datetime.date(1970, 1, 2),
+            None,
+            datetime.date(1981, 10, 25),
+            None,
+            datetime.date(2012, 5, 18),
+            None,
+            datetime.date(2019, 9, 3),
+            None,
+        ]
+    else:
+        dates = [
+            datetime.date(1970, 1, 1),
+            datetime.date(1970, 1, 2),
+            datetime.date(1981, 10, 25),
+            datetime.date(2012, 5, 18),
+            datetime.date(2019, 9, 3),
+        ]
```

Comment:
I agree that's a reasonable way of handling the nulls, however I feel like it increases the cognitive burden placed upon future readers of the code who need to mentally unpack the chain/zip to figure out what the code is doing.

How about this:
```
   if nullable:
        # Insert some None values in no particular order.  These will get
        # converted into avro "nulls" by the fastavro writer.
        dates = [
            None,
            datetime.date(1970, 1, 1),
            datetime.date(1970, 1, 2),
            datetime.date(1981, 10, 25),
            None,
            None,
            datetime.date(2012, 5, 18),
            None,
            datetime.date(2019, 9, 3),
            None,
        ]
    else:
        ...
```

Reply:
I guess my query/worry is are the dates for this test that you chose special in some way? e.g. 1970-01-01 certainly is in epoch terms. If so, repeating that actual dates is a mild DRY antipattern. Instead of the chain, how about making a list with `None` dates by default and filtering out in the other case
```python
dates = ... # with None
if not nullable:
    dates = list(filter(lambda: d is not None, dates))
```
?

Reply:
Ah!  I see your point now.  Yeah that's very reasonable, fixed.  Pushing shortly.

Reply:
This is what I'm going with:

```
-    if nullable:
-        # Insert some None values in no particular order.  These will get
-        # converted into avro "nulls" by the fastavro writer.
-        dates = [
-            None,
-            datetime.date(1970, 1, 1),
-            datetime.date(1970, 1, 2),
-            datetime.date(1981, 10, 25),
-            None,
-            None,
-            datetime.date(2012, 5, 18),
-            None,
-            datetime.date(2019, 9, 3),
-            None,
-        ]
-    else:
-        dates = [
-            datetime.date(1970, 1, 1),
-            datetime.date(1970, 1, 2),
-            datetime.date(1981, 10, 25),
-            datetime.date(2012, 5, 18),
-            datetime.date(2019, 9, 3),
-        ]
+    # Insert some None values in no particular order.  These will get converted
+    # into avro "nulls" by the fastavro writer (or filtered out if we're not
+    # nullable).  The first and last dates are epoch min/max values, the rest
+    # are arbitrarily chosen.
+    dates = [
+        None,
+        datetime.date(1970, 1, 1),
+        datetime.date(1970, 1, 2),
+        datetime.date(1981, 10, 25),
+        None,
+        None,
+        datetime.date(2012, 5, 18),
+        None,
+        datetime.date(2019, 9, 3),
+        None,
+        datetime.date(9999, 12, 31),
+    ]
+
+    if not nullable:
+        dates = [date for date in dates if date is not None]
```

Reply:
:+1:  (I have written too much haskell so I always reach for the functional combinators rather than list comprehensions).